src/EditModel/SectionDialog.component.js

```diff
@@ -359,6 +361,17 @@ class SectionDialog extends React.Component {
         });
     };
 
+    handleSectionTextUpdate = (fieldName) => (ev, text) => {
+        const cleanHtml = DOMPurify.sanitize(text);
```

Comment:
Should we call this in `componentWillReceiveProps` as well? So that the initial-value is sanitized as well? 

Of course this should ideally be done server-side, but this could help if someone opened a section that has been changed through API or some other means (and don't change the text).

Maybe we should extract it to it's own method?

```js

sanitizeDisplayOptions(displayOptions) //could be static or outside, idk

// in this code:

const newDisplayOption = sanitizeDisplayOptions({ displayOptions: { ...this.state.displayOptions, [fieldName]: cleanHtml }})

// componentWillReceiveProps

  displayOptions: props.sectionModel.displayOptions
                        ? this.sanitizeDisplayOptions(JSON.parse(props.sectionModel.displayOptions))
                        : {},

```

If you think it's unnecessary to always sanitize both keys, I'm fine with also sanitizing them directly.

```js
sanitizeText(text)

const newDisplayOptions = {
...this.state.displayOptions,
[fieldName]: sanitize(text)
}

// would make componentWillReceiveProps handling a bit more convoluted though

```

Just a bit more convenient to pass the entire thing.

Reply:
Disregard my last comment. I think it would be better to just run sanitization in `saveSection`, right?